neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,182 @@
+import logging
+from typing import Dict
+import numpy as np
+import math
+
+import torch
+import torch.nn as nn
+
+from neuralhydrology.modelzoo.fc import FC
+from neuralhydrology.modelzoo.head import get_head
+from neuralhydrology.modelzoo.basemodel import BaseModel
+
+LOGGER = logging.getLogger(__name__)
+
+class Transformer(BaseModel):
+    """Transformer model class, which relies on PyTorch's TransformerEncoder class.
+    This class implements the encoder of a transformer network which can be used for regression.
+    It is necessary to use an embedding network on all inputs, and the final embedding dimension (embedding_hiddens) 
+    must be divisible by the number of transformer heads (transformer_nheads).
+    The model configuration is specified in the config file using the following options:
+    -- transformer_positional_encoding_type : choices to "sum" or "concatenate" positional encoding to other model inputs.
+    -- transformer_positional_dropout: fraction of dropout applied to the positional encoding.
+    -- seq_length : integer number of timesteps to treat in the input sequence.
+    -- transformer_nheads : number of self-attention heads.
+    -- transformer_dim_feedforward : dimension of the feed-forward networks between self-attention heads.
+    -- transformer_dropout: dropout in the feedforward networks between self-attention heads.
+    -- transformer_nlayers: number of stacked self-attention + feedforward layers.
```

Comment:
```suggestion
    """Transformer model class, which relies on PyTorch's TransformerEncoder class.

    This class implements the encoder of a transformer network which can be used for regression.
    It is necessary to use an embedding network on all inputs, and the final embedding dimension (``embedding_hiddens``) 
    must be divisible by the number of transformer heads (``transformer_nheads``).
    The model configuration is specified in the config file using the following options:

    * ``transformer_positional_encoding_type``: choices to "sum" or "concatenate" positional encoding to other model
      inputs.
    * ``transformer_positional_dropout``: fraction of dropout applied to the positional encoding.
    * ``seq_length``: integer number of timesteps to treat in the input sequence.
    * ``transformer_nheads``: number of self-attention heads.
    * ``transformer_dim_feedforward``: dimension of the feed-forward networks between self-attention heads.
    * ``transformer_dropout``: dropout in the feedforward networks between self-attention heads.
    * ``transformer_nlayers``: number of stacked self-attention + feedforward layers.
```
otherwise the list won't render properly in readthedocs.